server/repository/repository.go

```diff
@@ -100,8 +100,11 @@ func (s *Server) ListApps(ctx context.Context, q *RepoAppsQuery) (*RepoAppsRespo
 
 	items := make([]*AppInfo, 0)
 
-	for i := range ksonnetRes.Items {
-		items = append(items, &AppInfo{Type: string(repository.AppSourceKsonnet), Path: ksonnetRes.Items[i]})
+	for _, i := range ksonnetRes.Items {
+		t := repository.IdentifyAppSourceTypeByAppDir(filepath.Dir(i))
```

Comment:
Does this code even work? Based on the implementation of IdentifyAppSourceTypeByAppDir (), you need direct access to the git repo on the local filesystem, which the API server does not.

Reply:
@jessesuen okay, I think we're a little better now. :D